src/jit/codegenxarch.cpp

```diff
@@ -4533,6 +4533,22 @@ void CodeGen::genStoreInd(GenTreePtr node)
                     assert(rmwSrc == data->gtGetOp2());
                     genCodeForShiftRMW(storeInd);
                 }
+                else if (data->OperGet() == GT_ADD && rmwSrc->IsIntegralConst(1))
+                {
+                    // Generate inc [mem] instead of "add [mem], 1".
+                    // Note that we don't need to check for GT_SUB of -1 because
+                    // global morph would transform it to GT_ADD of 1.
+                    assert(rmwSrc->isContainedIntOrIImmed());
```

Comment:
I know, but in this case it should generate sub x, -1/1 in the context of a RMW operation.  I have obtained SuperPMI difs with additional check for GT_SUB but it didn't produce any additional diffs. When we find real world examples producing such a code sequence, we can always go back and modify this condition.